unityfiles/Assets/Scripts/Game/Menu/PauseMenu.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PauseMenu : MonoBehaviour {
+    [SerializeField]
+    private GameObject pauseMenuCanvas;
+
+    [SerializeField]
+    private GameObject confirmDialog;
+
+    private Boolean isPaused = false;
+
+    private Button continueButton;
+    private static readonly int PAUSE = 0;
+    private static readonly float PLAY = 1;
+
+    private void Awake() {
+        CheckForMissingComponents();
+        
+        DeactivateMenu(); 
+    }
+
+    private void Start() {
+        SubscribeToEvents();
+    }
+
+    // Update is called once per frame
+    void Update() {
+        if (Input.GetButtonDown("Cancel")) {
+            if (confirmDialog.activeSelf) {
+                DeactivateConfirmDialog();
+            } else {
+                TogglePause();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Toggles the pause state.
+    /// Shows pause menu if paused
+    /// </summary>
+    private void TogglePause() {
+        isPaused = !isPaused;
+
+        if (isPaused) {
+            PauseGame();
+        } else {
+            ResumeGame();
+        }
+    }
+
+    private void OnDestroy() {
+        UnsubscribeFromEvents();
+    }
+
+    private void UnsubscribeFromEvents() {
+        LevelManager.LevelStateChangeEvent -= ChangeGameState();
+    }
+
+    private void SubscribeToEvents() {
+        LevelManager.LevelStateChangeEvent += ChangeGameState();
+    }
+
+    private EventHandler<LevelStateChangeEventArgs> ChangeGameState() {
+        return (sender, args) => {
+            if (args.NewState == LEVEL_STATE.PAUSE) {
+                PauseGame();
+            }
+
+            if (args.NewState == LEVEL_STATE.RESUME) {
+                if (confirmDialog.activeSelf) {
+                    DeactivateConfirmDialog();
+                } else {
+                    ResumeGame();
+                }
+            }
+
+            if (args.NewState == LEVEL_STATE.CONFIRM) {
+                ActivateConfirmDialog();
+            }
+        };
+    }
+
+    private void DeactivateConfirmDialog() {
+        confirmDialog.SetActive(false);
+    }
+
+    private void ActivateConfirmDialog() {
+        confirmDialog.SetActive(true);
+    }
```

Comment:
persjonelig syen enlinjes formler som igjenn bare kaller en fungsjon på noe er unødvendige clutter. hvis du skal ha de kommenter i allefall

Reply:
Jeg har gjort dette av to grunner. Den ene er for lesbarhet, den andre er fordi jeg under utvikling har gjort flere ting inni disse metodene, men tatt det ut underveis.
Siden jeg synes metodene er såpass leslige har jeg valgt å la dem stå. Vi kan jo ta en prat om dette også i felleskap